### DISCUSSION TITLE
Sync transactions from Lunch Money to Splitwise

### DISCUSSION THREAD

**Discussion by danielr18** [juftin/lunchable#11] [Ideas]
> Goal: Allows me to add a Lunch Money transaction directly to Splitwise (via a Tag or something like that) which splits the transaction in the process

Hi @juftin,

We can use discussions to talk about how certain features could be implemented if you'd like.

Is the idea to keep `https://github.com/juftin/lunchable/blob/feature/plugin/splitlunch/poc/lunchable/plugins/splitlunch/lunchmoney_splitwise.py` mostly as a splitwise API wrapper and then let users create their own scripts to achieve the goal? Something similar to [this function](https://gist.github.com/danielr18/66bf4133632100a20cf1ef6f60a2d3fe#file-lunchmoneysplitwise-js-L23) but using the lunchable api and the plugin.

Example:
```
transactions = get transactions from lunchmoney that match filters
for transaction in transactions:
  expense = convert TransactionObject to SplitLunchExpense
  add_expense_to_splitwise() 
```

**Comment by juftin** [comment]
@danielr18 great question. I have a few goals for this plugin. The below 3 items I have envisioned as an always-on service that runs in the background - but ability for users to build their own scripts is definitely a motivation.

1) Non self-paid transactions (transactions where you owe money to other people) originating in Splitwise get auto-imported into Lunch Money
2) Some kind of tagging routine inside of Lunch Money (i.e. expenses with a `SplitLunch` tag) that creates self-paid Splitwise transactions. My current idea for this is to designate a `"Financial Partner"` (a linked friend in Splitwise) who the expense gets split with. This functionality will also need to split the original Lunch Money transaction and place one half of the split under a category called `Reimbursed` so it's excluded from Monthly Totals. 
3) Syncing of Splitwise Balance (with the above "Financial Partner") to a manually managed Lunch Money Asset

Those functionalities sort of outline my own personal mindset around factoring Splitwise into my personal finances and might not make sense for everyone's use cases. I'm interested to hear your thoughts and if you would think of doing it differently.
